src/fps.rs

```diff
@@ -1,21 +1,52 @@
 use std::future::Future;
-use std::sync::atomic::{AtomicU32, Ordering};
-use std::time::Duration;
+use std::time::{Duration, Instant};
 
 use dcs_module_ipc::IPC;
+use once_cell::sync::Lazy;
 use stubs::mission::v0::stream_events_response::{Event, SimulationFpsEvent};
 use stubs::mission::v0::StreamEventsResponse;
+use tokio::sync::Mutex;
 use tokio::time::{interval, MissedTickBehavior};
 
-static FPS: AtomicU32 = AtomicU32::new(0);
-static TIME: AtomicU32 = AtomicU32::new(0);
+static FPS: Lazy<Mutex<Option<Fps>>> = Lazy::new(Default::default);
 
-pub fn frame(time: f64) {
-    // Increase the frame count by one
-    FPS.fetch_add(1, Ordering::Relaxed);
+struct Fps {
+    /// The DCS simulation time of the last update.
+    simulation_time: f64,
+    /// The DCS real time of the last update.
+    real_time: f64,
+    /// The instant the previous frame got measured.
+    previous: Instant,
+    /// Collection of frame times in milliseconds.
+    frames_in_dcs_time: Vec<f32>,
+    /// Collection of frame times in milliseconds.
+    frames_in_rust_time: Vec<f32>,
+}
+
+pub fn frame(simulation_time: f64, real_time: f64) {
+    let mut fps = FPS.blocking_lock();
```

Comment:
As this change requires a `Vec`, I don't see a way around a mutex anymore. I'd guess that this might take up to `0.002ms`  to get a lock for each simulation frame (I don't really have an idea, so this is a very wild guess based on some mutex stats I've glanced at).